File: src/main/java/client/database/data/BbsThreadData.java
package client.database.data;

import java.util.ArrayList;
import java.util.List;

public class BbsThreadData {
   private int posterCharacterId;

   private long timestamp;

   private String name;

   private String startPost;

   private int icon;

   private int replyCount;

   private int threadId;

   private List<BbsThreadReplyData> replyDataList;

   public BbsThreadData(int posterCharacterId, long timestamp, String name, String startPost, int icon, int replyCount, int threadId) {
      this.posterCharacterId = posterCharacterId;
      this.timestamp = timestamp;
      this.name = name;
      this.startPost = startPost;
      this.icon = icon;
      this.replyCount = replyCount;
      this.threadId = threadId;
   }

   public void addReply(BbsThreadReplyData replyData) {
      if (replyDataList == null) {
         replyDataList = new ArrayList<>();
      }
      replyDataList.add(replyData);
   }

   public List<BbsThreadReplyData> getReplyData() {
      return replyDataList;
   }

   public int getPosterCharacterId() {
      return posterCharacterId;
   }

   public long getTimestamp() {
      return timestamp;
   }

   public String getName() {
      return name;
   }

   public String getStartPost() {
      return startPost;
   }

   public int getIcon() {
      return icon;
   }

   public int getReplyCount() {
      return replyCount;
   }

   public int getThreadId() {
      return threadId;
   }
}

File: engine/model/src/main/scala/client/database/data/BbsThreadData.scala
package client.database.data

import scala.jdk.CollectionConverters._

class BbsThreadData(private var _posterCharacterId: Int, private var _timestamp: Long, private var _name: String,
                    private var _startPost: String, private var _icon: Int, private var _replyCount: Int,
                    private var _threadId: Int) {
  def posterCharacterId: Int = _posterCharacterId

  def timestamp: Long = _timestamp

  def name: String = _name

  def startPost: String = _startPost

  def icon: Int = _icon

  def replyCount: Int = _replyCount

  def threadId: Int = _threadId

  private var replyDataList: List[BbsThreadReplyData] = List[BbsThreadReplyData]()

  def addReply(bbsThreadReplyData: BbsThreadReplyData): Unit = {
    replyDataList = replyDataList.appended(bbsThreadReplyData)
  }

  def getReplyData: java.util.List[BbsThreadReplyData] = {
    replyDataList.asJava
  }
}
